mako/util.py

```diff
@@ -166,7 +166,24 @@ def getvalue(self):
                 self.encoding, self.errors
             )
         else:
-            return self.delim.join(self.data)
+            try:
+                return self.delim.join(self.data)
+            except:
+                # these exceptions are usually because of an encoding issue
+                # the standard exception is vague because it happens within a
+                # line but does not show the exact line.
+                try:
+                    for idx in range(0, len(self.data)):
+                        _content = self.delim.join(list(self.data)[0:idx+1])
```

Comment:
you can do `list(self.data)` before the loop.

Also why not using a binary search?

Reply:
> Also why not using a binary search?

this is probably overkill for what this code aims to do, so nevermind that :)